website/www/site/content/en/documentation/sdks/yaml.md

```diff
@@ -366,6 +366,8 @@ pipeline:
       config:
         topic: anotherPubSubTopic
         format: json
+options:
+  streaming: true
```

Comment:
This should be implicit due to the PubSub source. I don't know that we want to recommend setting it everywhere (though +1 to the section talking about options in the docs).

Reply:
If I remember correctly, I've had issues with Pub/Sub pipelines when the `streaming` option is not set. It could be an issue with xlang, but I agree it *should* be implicit, so it may be worth looking into fixing on the xlang side rather than explicitly defining in the YAML config.

Reply:
For python it is never implicit IIRC (we don't check whether sources are bounded or unbounded), so unless we're doing something special with yaml then I think we need this.

@tvalentyn to fact check me here, but I think we talked about making streaming implicit for Python but decided against it for back-compat reasons

Reply:
Making it implicit should not be backwards incompatible, as otherwise the pipeline wouldn't run, right? 

If this is the case (we should confirm), I guess it's better to add it until the template is updated with the fix at the very least.

Reply:
> Making it implicit should not be backwards incompatible, as otherwise the pipeline wouldn't run, right?

The pipeline will just try to run in batch mode, which is actually probably fine for some unbounded sources (I think). It will run in batch mode, but generally functions fine AFAIK.

With that said, I don't really fully remember what our concerns with opting in to streaming first mode are. There are use cases here ([example](https://lists.apache.org/thread/p9g47xfowygjdrjjwo2f62hfjdjy70s7)), though they're pretty dubious IMO.

> I don't recall. What's the behavior in other SDKs?

Other SDKs opt in any unbounded sources to streaming pipelines.

Reply:
it makes sense to have a consistent behavior across SDKs. If we consider making the change, we should run some queries re: # of pipelines that had a pubsub step and were not streaming. most of those if any might be launched in error? But if there _are_ users who intentionally run such jobs, we need to see if they will have an opt-out mechanism from new behavior.. There is a --streaming option, but no --batch option.

Reply:
To be clear, this isn't just for pubsub, this is for any pipeline with an unbounded source

Reply:
https://github.com/apache/beam/issues/30531

Reply:
Thanks